File: prak2 java/Hardware.java
class Hardware extends Product{
    
    private String brand;
    private String model;

    Hardware(){
        //konstruktor
    }

    Hardware(Product x, String brand, String model){
        //konstruktor
        this.setPrice(x.getPrice());
        this.setIdProduct(x.getIdProduct());
        this.brand = brand;
        this.model = model;
    }

    Hardware(int price, String idProduct, String brand, String model){
        //konstruktor
        this.setPrice(price);
        this.setIdProduct(idProduct);
        this.brand = brand;
        this.model = model;
    }

    //get set
    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    //methods
    public void tampilkanHardware(){
        System.out.println("Price: " + getPrice());
        System.out.println("idProduct: " + getIdProduct());
        System.out.println("Brand: " + getBrand());
        System.out.println("Model: " + getModel());
    }

}

File: prak2 php/Hardware.php
<?php

class Hardware extends Product{

    private $brand;
    private $model;

    function __construct($product, $br, $mo){
        //konstruktor
        $this->setPrice($product->getPrice());
        $this->setIdProduct($product->getIdProduct());
        $this->brand = $br;
        $this->model = $mo;
    }


    //set
    public function setBrand($br){
        $this->brand = $br;
    }

    public function setModel($mo){
        $this->model = $mo;
    }

    //get
    public function getBrand(){
        return $this->brand;
    }

    public function getModel(){
        return $this->model;
    }


    //methods
    public function tampilkanHardware(){
        echo "Price : ".$this->getPrice()."<br/>";
        echo "idProduct : ".$this->getIdProduct()."<br/>";
        echo "Brand : ".$this->getBrand()."<br/>";
        echo "Model : ".$this->getModel()."<br/>";
    }


    function __destruct(){
        //destruktor
    }

}

?>
